packages/design-system/src/components/Toggle/__tests__/Toggle.screen-object.tsx

```diff
@@ -0,0 +1,52 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { useState } from "react";
+import Toggle from "..";
+import { ToggleProps } from "../Toggle.interface";
+
+export const ToggleTestComponent = ({
+  ...props
+}: Omit<ToggleProps, "checked" | "onChange" | "value">) => {
+  const [checked, setChecked] = useState<boolean>(false);
+  return (
+    <>
+      <Toggle
+        {...props}
+        checked={checked}
+        onChange={() => setChecked(!checked)}
```

Comment:
```suggestion
        onChange={() => setChecked(prevState => !prevState)}
```
Using this version of changing state, you are sure that the result is the expected result.